### PR TITLE
Drop support for Python <3.10

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [ ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [ ] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [ ] I've described my changes in the section below.

### Changes description
Removes Python <3.10 from testing making 3.10 the minimal version we use for development.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,12 +10,12 @@ With Snowflake CLI, developers can create, manage, update, and view apps running
 
 ## Install Snowflake CLI
 
-### Install with pip (PyPi)
+### Install with pipx (PyPi)
 
-Requires Python >= 3.8
+We recommend installing Snowflake CLI in isolated environment using [pipx](https://pipx.pypa.io/stable/). Requires Python >= 3.10
```
Comment (sfc-gh-astus) [inline README.md:15]: Requires Python >= 3.11

File: README.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ snow --help
 
 ### Install from source
 
-Requires Python >= 3.8 and git
+Requires Python >= 3.10 and git
```
Comment (sfc-gh-astus) [inline README.md:34]: 3.11

File: README.md (hunk 1 above)
Comment (sfc-gh-turbaszek) [inline README.md:15]:
```suggestion
We recommend installing Snowflake CLI in isolated environment using [pipx](https://pipx.pypa.io/stable/). Requires Python >= 3.11
```

File: README.md (hunk 2 above)
Comment (sfc-gh-turbaszek) [inline README.md:34]:
```suggestion
Requires Python >= 3.11 and git
```

**Review by sfc-gh-astus — DISMISSED** [review]

**Comment by sfc-gh-astus** [comment]
MAybe add this info to release-notes?

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**